### PR TITLE
Update API doc for ActiveRecord::Relation#none?/#any?/#one? [ci-skip]

### PR DESCRIPTION
**Pull request by mechanicles**
### Motivation / Background

I checked https://github.com/rails/rails/issues/50327 and noticed that clear API docs are needed. Some developers are using these methods incorrectly, thinking they're only for Active Record Relation and pass condition argument, but they actually work with Enumerable. These methods treat that argument as a pattern argument.

### Detail

I have added required API doc for these methods so developers won't get confused with the argument meaning.

### Additional information

Original [PR](https://github.com/rails/rails/pull/46728) which updated these methods to support pattern argument.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] [N/A] Tests are added or updated if you fix a bug or add a feature.
* [x] [N/A]  CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by p8** [comment]
Thanks @mechanicles !
Adding an example could help clarify things. 
Maybe use the examples in the test cases of the linked PR?

**Comment by mechanicles** [comment]
> Thanks @mechanicles ! Adding an example could help clarify things. Maybe use the examples in the test cases of the linked PR?

@p8 I have added required examples in the code API documentations. Please let me know If I missed something.

File: activerecord/lib/active_record/relation.rb [hunk 1]
```diff
@@ -291,6 +291,11 @@ def empty?
     end
 
     # Returns true if there are no records.
+    #
+    # When a pattern argument is given, this method checks whether elements in
+    # the Enumerable match the pattern via the case-equality operator, `===`.
```
Comment (p8) [inline activerecord/lib/active_record/relation.rb:296]: Some nitpicks about the punctuation.
I'm not sure a comma is correct here, but I could be wrong (not a native English speaker).

A colon doesn't look right either.
```suggestion
    # the Enumerable match the pattern via the case-equality operator: `===`.
```

Maybe:
```suggestion
    # the Enumerable match the pattern via the case-equality operator (`===`).
```

Comment (fatkodima) [reply activerecord/lib/active_record/relation.rb:296]: +1 for the parentheses version

Comment (mechanicles) [reply activerecord/lib/active_record/relation.rb]: Yes. It makes. I updated with the parentheses version.

**Comment by p8** [comment]
Thanks @mechanicles !

**Pull request merged**